dotCMS/src/main/java/com/dotmarketing/util/FileUtil.java

```diff
@@ -234,6 +242,78 @@ public boolean accept(File pathname) {
             }
         };
     }
+
+	/**
+	 * Figure out the sha256 of the file content, assumes that the file exists and can be read
+	 * @param file {@link File}
+	 * @return String  just as unix sha returns
+	 * @throws NoSuchAlgorithmException
+	 * @throws IOException
+	 */
+	public static String sha256toUnixHash (final File file) throws NoSuchAlgorithmException, IOException {
+
+		return sha256toUnixHash(file.toPath());
+	} // sha256toUnixHash.
+
+	/**
+	 * Figure out the sha256 of the file content, assumes that the file exists and can be read
+	 * @param path {@link Path}
+	 * @return String just as unix sha returns
+	 * @throws NoSuchAlgorithmException
+	 * @throws IOException
+	 */
+    public static String sha256toUnixHash(final Path path) throws NoSuchAlgorithmException, IOException {
+
+		final HashBuilder sha256Builder = Encryptor.Hashing.sha256();
+		final byte[] buffer             = new byte[BUFFER_SIZE];
+		int countBytes 					= 0;
+
+		try (InputStream inputStream = new BufferedInputStream(Files.newInputStream(path))) {
+
+			countBytes = inputStream.read(buffer);
+			while (countBytes > 0) {
+
+				sha256Builder.append(buffer, countBytes);
+				countBytes = inputStream.read(buffer);
+			}
+		}
+
+		return sha256Builder.buildUnixHash();
+	} // sha256toUnixHash.
+
+	/**
+	 * Given an absolute path this will figure out if it exists under the real assets and return the relative piece
+	 * @param file
+	 * @return
+	 */
+	@VisibleForTesting
+	static Optional<String> getRealAssetsPathRelativePiece(final File file,
```

Comment:
We don't need two arguments here - we know the realAssetsPath system wide.  Actually, I think we really don't think we need these two methods in here at all.  why not just do a:

```
String relPath = file.getAbsolutePath().replaceAll(ConfigUtils.getAbsoluteAssetsRootPath());
```

Reply:
ok ok